RustBoy/RustBoy.swift

```diff
@@ -7,103 +7,147 @@
 //
 
 #if os(OSX)
-import RustBoy
+import AppKit
+import CoreRustBoy
+import CoreVideo
 #endif
 
 internal class RustBoy {
 
 	internal enum ButtonType {
 		case left, right, up, down, a, b, start, select
-#if os(OSX)
-		fileprivate var asCoreButton: rustBoyButton {
-			switch self {
-				case .left:		return rustBoyButtonLeft
-				case .right:	return rustBoyButtonRight
-				case .up:		return rustBoyButtonUp
-				case .down:		return rustBoyButtonDown
-				case .a:		return rustBoyButtonA
-				case .b:		return rustBoyButtonB
-				case .start:	return rustBoyButtonStart
-				case .select:	return rustBoyButtonSelect
-			}
-		}
-#endif
 	}
 
-	internal struct Cartridge {
-		fileprivate let path: String
-		fileprivate let saveFilePath: String?
-
-		internal init(path: String, saveFilePath: String? = nil) {
-			self.path = path
-			self.saveFilePath = saveFilePath
-		}
+    internal struct Cartridge: Equatable {
+		internal let path: String
+		internal let saveFilePath: String
 	}
 
 	internal enum BootStatus: Error {
-		case propertyMissing(name: String)
+        case cartridgeMissing
 		case failedToInitCore
 		case success
 	}
 
 	internal var cartridge: Cartridge? {
 		didSet {
-			boot()
+			let result = boot()
+            assert(result == .success, "Boot failed with error: \(result)")
 		}
 	}
 
-	internal var display: DisplayView? {
-		didSet {
-			boot()
-		}
-	}
+    internal var display: DisplayView? {
+        didSet {
+#if os(OSX)
+            coreRustBoy?.display = display
+#endif
+        }
+    }
 
 	internal init() {}
 
-	internal func buttonDown(_ button: ButtonType) {
-		guard let coreRef = coreRef else { return }
+    internal func buttonDown(_ button: ButtonType) {
 #if os(OSX)
-		rustBoyButtonClickDown(coreRef, button.asCoreButton)
+        coreRustBoy?.buttonDown(rustBoyButton(button))
```

Comment:
I kinda feel this should be buttonPressed and buttonUnpressed or something? buttonDown can become confused with the Down button.